bot/cogs/reddit.py

```diff
@@ -63,23 +74,18 @@ def __init__(self, bot: Bot):
         log.debug(f"Invalid response from: {url} - status code {response.status}, mimetype {response.content_type}")
         return list()  # Failed to get appropriate response within allowed number of retries.
 
-    async def send_top_posts(
-        self, channel: TextChannel, subreddit: Subreddit, content: str = None, time: str = "all"
-    ) -> Message:
-        """Create an embed for the top posts, then send it in a given TextChannel."""
+    async def get_top_posts(self, subreddit: Subreddit, time: str = "all") -> Embed:
+        """Returns an embed for the top posts of the given subreddit."""
         # Create the new spicy embed.
-        embed = Embed()
-        embed.description = ""
-
-        # Get the posts
-        async with channel.typing():
-            posts = await self.fetch_posts(
-                route=f"{subreddit}/top",
-                amount=5,
-                params={
-                    "t": time
-                }
-            )
+        embed = Embed(description="")
+
+        posts = await self.fetch_posts(
+            route=f"{subreddit}/top",
+            amount=5,
+            params={
+                "t": time
+            }
```

Comment:
splitting this into three lines seems overkill. a one-liner would probably yield better readability here.